### PR TITLE
HW-06-ready

### PR DESCRIPTION
**Pull request by YaroslavKuzero**

### CODE REVIEW

File: api/auth/auth.controller.js [hunk 1]
```diff
@@ -17,11 +19,14 @@ const registerController = async (req, res) => {
...
       })
     }
+    const verificationToken = uuidv4();
     const cryptedPassword = await bcrypt.hash(body.password, SALT);
     const newUser = await User.createUser({
       ...body,
       password: cryptedPassword,
+      verificationToken,
     });
+    sendEmail(newUser.email, verificationToken);
```
Comment (odykyi) [inline api/auth/auth.controller.js]: await

File: services/email.service.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const sgMail = require('@sendgrid/mail');
+const { config } = require('./config')
+
+sgMail.setApiKey(process.env.SENDGRID_API_KEY);
+
+const msg = (receiver, token) => {
+  return {
+    to: receiver,
+    from: 'yarikkuzero@gmail.com',
```
Comment (odykyi) [inline services/email.service.js]: move from to config

File: services/email.service.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    text: "Let's verify your email. Click link below",
+    html: `<a href='${config.hostURL}auth/verify/${token}'>Start to use our awesome app</a>`
+  }
+}
+
+const sendEmail = async (receiver, token) => {
+  const message = await msg(receiver, token);
+  console.log(message);
+  try {
+    await sgMail.send(message).then(() => console.log('Email sent'))
```
Comment (odykyi) [inline services/email.service.js]: remove then

**Review by odykyi — CHANGES_REQUESTED** [review]
comments

File: services/jwt.service.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 const jwt = require('jsonwebtoken');
-const KEY = "kayofmcvxperin5dl9qpwb";
+const KEY = process.env.KEY;
```
Comment (odykyi) [inline services/jwt.service.js]: KEY - its implicit

use JWT_SECRET_KEY

File: services/config.js [hunk 5]
```diff
@@ -15,6 +15,7 @@ const config = {
   hostURL: 'http://localhost:3000/',
```
Comment (odykyi) [inline services/config.js:15]: remove async await  from functions createAvaURL, tempPath and createAvaDB

Comment (odykyi) [reply services/config.js:15]: @YaroslavKuzero

**Review by odykyi — CHANGES_REQUESTED** [review]
comments

File: services/jwt.service.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 const jwt = require('jsonwebtoken');
-const KEY = "kayofmcvxperin5dl9qpwb";
+const JWT_SECRET_KEY = process.env.JWT_SECRET_KEY;
 
 const createToken = async (payload) => {
   const token = await jwt.sign(payload, KEY);
```
Comment (odykyi) [inline services/jwt.service.js]: use JWT_SECRET_KEY instead of KEY

**Review by odykyi — CHANGES_REQUESTED** [review]
comments

File: services/jwt.service.js (hunk 6 above)
Comment (YaroslavKuzero) [reply services/jwt.service.js]: Sorry) Misprint

**Review by odykyi — APPROVED** [review]

**Pull request merged**